Error checking for texture loading to fix green video on Linux (#647)

johncurley opened the pull request:
An uncaught error causes the game intro video to not load correctly due to a false positive on Linux and MacOS platforms. This fix allows the error stack to be popped before it reaches the error statement on the Linux platform.

johncurley committed:
added error checking for textures on release builds

375gnu commented on src/Layers/xrRenderGL/glTexture.cpp:
```diff
@@ -121,21 +121,31 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
         case gli::TARGET_CUBE:
             CHK_GL(glTexStorage2D(target, static_cast<GLint>(texture.levels()), format.Internal,
                            tex_extent.x, tex_extent.y));
+            if ((err = glGetError()) != GL_NO_ERROR)
```
Remove CHK_GL to make this work (CHK_GL uses glGetError())

johncurley replied:
Isn't CHK_GL active only on Debug builds? This is happening on release builds.

375gnu replied:
I believe it's better to have a consistent behavior in all builds.

Xottab-DUTY commented on src/Layers/xrRenderGL/glTexture.cpp:
```diff
@@ -121,21 +121,31 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
+            if ((err = glGetError()) != GL_NO_ERROR)
+            {
+                Msg("Error: 0x%x: Invalid 3D texture: '%s'", err, fname);
+            }
```
Here's what I suggest:
```suggestion
            glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
                           tex_extent.x, tex_extent.y, tex_extent.z);
            const auto err = glGetError();
            if (err != GL_NO_ERROR)
            {
                VERIFY(err == GL_NO_ERROR, "Error when loading a texture");
                Msg("! OpenGL: 0x%x: Invalid 3D texture: '%s'", err, fname);
            }
```

johncurley committed:
Update src/Layers/xrRenderGL/glTexture.cpp

CHK_GL is not sufficient for covering both release and debug modes so we have decided to remove it and replace with another error checking mechanism.

Co-authored-by: Xottab_DUTY <Xottab-DUTY@users.noreply.github.com>

johncurley committed:
Merge branch 'OpenXRay:dev' into green-video-linux-fix

johncurley committed:
removed CHK_GL and replaced with error message

johncurley committed:
Cleaned up VERIFY statements

Xottab-DUTY commented on src/Layers/xrRenderGL/glTexture.cpp:
```diff
@@ -6,6 +6,9 @@
 
+//#include "xrCore/xrDebug_macros.h"
+
+
```
```suggestion
```

Xottab-DUTY commented on src/Layers/xrRenderGL/glTexture.cpp:
```diff
@@ -121,21 +124,35 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
         case gli::TARGET_3D:
-            CHK_GL(glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
-                           tex_extent.x, tex_extent.y, tex_extent.z));
+        case gli::TARGET_CUBE_ARRAY:
```
```suggestion
```

Xottab-DUTY commented on src/Layers/xrRenderGL/glTexture.cpp:
```diff
@@ -160,35 +177,61 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
                         break;
                     }
                     case gli::TARGET_3D:
+                    case gli::TARGET_CUBE_ARRAY:
```
```suggestion
```

Xottab-DUTY committed:
Apply suggestions from code review
